ppmi_metadata/ppmi_metadata.py

```diff
@@ -36,6 +36,101 @@ def __init__(self, email, password):
+        '''
+
+        subjectIds = ','.join([str(i) for i in subject_ids])
+
+        # Create Chrome webdriver
+        options = webdriver.ChromeOptions()
+        tempdir = op.abspath(tempfile.mkdtemp(dir='.'))
+        prefs = {'download.default_directory': tempdir}
+        options.add_experimental_option("prefs", prefs)
+        if headless:
+            options.add_argument("--headless")
+
+        self.driver = webdriver.Chrome(ChromeDriverManager().install(), chrome_options=options)
+
+        # Login to PPMI
+        self.driver.get('https://ida.loni.usc.edu/login.jsp?project=PPMI')
+        self.html = HTMLHelper(self.driver)
+        self.html.click_button("//div[contains(@class,'ida-cookie-policy-accept')]")
+        self.html.click_button("//div[contains(@class,'ida-user-menu-icon')]")
+        self._login(self.email, self.password)
+
+        # navigate to search page
+        self.driver.get('https://ida.loni.usc.edu/home/projectPage.jsp?project=PPMI')
+        self.html.click_button("//a[text()='SEARCH']")
+        self.html.click_button("//a[text()='Advanced Image Search (beta)']")
+        time.sleep(2)
+
+        # Enter id's and add to collection
+        self.html.enter_data('//*[@id="subjectIdText"]', subjectIds)
+        time.sleep(2)
+        self.html.click_button('//*[@id="advSearchQuery"]')
+        time.sleep(2)
+        self.html.click_button('//*[@id="advResultSelectAll"]')
+        time.sleep(3)
+        self.html.click_button('//*[@id="advResultAddCollectId"]')
+        time.sleep(2)
+        self.html.enter_data('//*[@id="nameText"]', "images")
+        time.sleep(3)
+        self.html.click_button('//*[text()="OK"]')
+        time.sleep(2)
+        self.html.click_button('//*[@id="selectAllCheckBox"]')
+        time.sleep(2)
+        self.html.click_button('//*[@id="simple-download-button"]')
+
+        # Download imaging data and metadata
+        self.html.click_button('//*[@id="simple-download-link"]')
+        time.sleep(2)
+        self.html.click_button(('//*[@class="simple-download-metadata-link-text singlefile-download-metadata-link"]'))
+
+        # Wait for download to complete
+        def download_complete(driver):
+            downloaded_files = os.listdir(tempdir)
+            assert(len(downloaded_files) <= 2)
+            if len(downloaded_files) == 0:
+                return False
+            f = downloaded_files[0]
+            if f.endswith('.crdownload'):
+                return False
+            assert(f.endswith('.csv') or f.endswith('.zip'))
+            return True
+        WebDriverWait(self.driver, timeout).until(download_complete)
+
+        # Move file to cwd or extract zip file
+        downloaded_files = os.listdir(tempdir)
+
+        # we got imaging data and metadata
+        assert(len(downloaded_files) == 2)
+
+        for file_name in downloaded_files:
+            assert(file_name.endswith('.zip') or file_name.endswith('.csv'))
```

Comment:
This assertion fails because files are .dcm

Reply:
Good catch